### DISCUSSION TITLE
editor.getHtml() not tracking changes

### DISCUSSION THREAD

**Discussion by srivatsan2607** [GrapesJS/grapesjs#4770] [Q&A]
Hi @artf , 
I hope you are doing well. 

I am making changes in the canvas directly by fetching the iframe and looping into the required element and changing certain things that are required. 

When I try to get the HTML code with the editor.getHtml() function, it returns the HTML code untracked of the changes that I made directly in the frame document. 
I think the editor is not tracking the current changes made to the document. Is there any possible way to explicitly track that change as well? 

Same case while storing as well.

Thank you for your help in advance.

**Comment by artf** [comment]
You should change [component](https://grapesjs.com/docs/modules/Components.html) itself in order to see updates in HTML code.

**Reply by srivatsan2607** [reply]
Okay, I understand. Is it possible to add the attributes like href for a tag and add innerHTMl to a component programattically?

**Reply by artf** [reply]
sure, you can use [addAttributes](https://grapesjs.com/docs/api/component.html#addattributes) and [components](https://grapesjs.com/docs/api/component.html#components).

**Reply by srivatsan2607** [reply]
Okay @artf, I will try it. Thank you.
